api/odl2.py

```diff
@@ -0,0 +1,260 @@
+            db,
+            collection,
+            parser,
+            data_source_name,
+            identifier_mapping,
+            http_get,
+            metadata_client,
+            content_modifier,
+            map_from_collection,
+            mirrors,
+        )
+
+        self._logger = logging.getLogger(__name__)
+
+        self._configuration_storage = ConfigurationStorage(self)
+        self._configuration_factory = ConfigurationFactory()
+
+    @contextmanager
+    def _get_configuration(self, db):
+        """Return the configuration object.
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :return: Configuration object
+        :rtype: ODL2APIConfiguration
+        """
+        with self._configuration_factory.create(
+            self._configuration_storage, db, ODL2APIConfiguration
+        ) as configuration:
+            yield configuration
+
+    def _extract_publication_metadata(self, feed, publication, data_source_name):
+        """Extract a Metadata object from webpub-manifest-parser's publication.
+
+        :param publication: Feed object
+        :type publication: opds2_ast.OPDS2Feed
+
+        :param publication: Publication object
+        :type publication: opds2_ast.OPDS2Publication
+
+        :param data_source_name: Data source's name
+        :type data_source_name: str
+
+        :return: Publication's metadata
+        :rtype: Metadata
+        """
+        metadata = super(ODL2Importer, self)._extract_publication_metadata(
+            feed, publication, data_source_name
+        )
+        formats = []
+        licenses = []
+        licenses_owned = 0
+        licenses_available = 0
+        medium = None
+
+        with self._get_configuration(self._db) as configuration:
+            skipped_license_formats = configuration.skipped_license_formats
+
+            if skipped_license_formats:
+                skipped_license_formats = set(json.loads(skipped_license_formats))
+
+        if publication.licenses:
+            for license in publication.licenses:
+                identifier = license.metadata.identifier
+
+                for license_format in license.metadata.formats:
+                    if (
+                        skipped_license_formats
+                        and license_format in skipped_license_formats
+                    ):
+                        continue
+
+                    if not medium:
+                        medium = Edition.medium_from_media_type(license_format)
+
+                    drm_schemes = (
+                        license.metadata.protection.formats
+                        if license.metadata.protection
+                        else []
+                    )
+
+                    if license_format == self.FEEDBOOKS_AUDIO:
+                        license_format = MediaTypes.AUDIOBOOK_MANIFEST_MEDIA_TYPE
+                        drm_schemes.append(DeliveryMechanism.FEEDBOOKS_AUDIOBOOK_DRM)
```

Comment:
This feels very specific. Should we accomplish this with a mapping to make this importer more generic?

Reply:
I added `ODL2Importer.LICENSE_FORMATS` mapping.